src/OpenConext/ProfileBundle/Security/Guard.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * Copyright 2015 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\ProfileBundle\Security;
+
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+final class Guard
+{
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     */
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
+    public function userIsLoggedIn()
+    {
+        if (!$this->authorizationChecker->isGranted('ROLE_USER')) {
+            throw new AccessDeniedHttpException('User should be logged in to use this service');
```

Comment:
I think we should either throw a [AccessDeniedException](https://github.com/symfony/symfony/blob/0055887cbdf89d9b3b0235f2089a2751bcb52d17/src/Symfony/Component/Security/Core/Exception/AccessDeniedException.php), which Symfony [converts](https://github.com/symfony/symfony/blob/0055887cbdf89d9b3b0235f2089a2751bcb52d17/src/Symfony/Component/Security/Http/Firewall/ExceptionListener.php#L119) to an AccessDeniedHttpException, or move the Guard to the namespace `...\Security\Http`.